templates/netlify/remix.init/index.js

```diff
@@ -0,0 +1,102 @@
+const inquirer = require("inquirer");
+const fs = require("fs/promises");
+const { join } = require("path");
+const PackageJson = require("@npmcli/package-json");
+
+const filesToCopy = [
+  ["README.md"],
+  ["netlify.toml"],
+  ["server.js"],
+  ["remix.config.js"],
+  ["vscode.json", join(".vscode", "settings.json")],
+  ["globals.d.ts"],
+];
+
+const filesToModify = ["app/entry.server.tsx", "app/root.tsx"];
+
+async function modifyFilesForEdge(files, rootDirectory) {
+  const filePaths = files.map((file) => join(rootDirectory, file));
+  const contents = await Promise.all(
+    filePaths.map((path) => fs.readFile(path, "utf8"))
+  );
+
+  await Promise.all(
+    contents.map((content, index) => {
+      const newContent = content.replaceAll(
+        "@remix-run/node",
+        "@remix-run/deno"
+      );
+      return fs.writeFile(filePaths[index], newContent);
+    })
+  );
+}
+
+async function copyEdgeTemplateFiles(files, rootDirectory) {
+  for (const [file, target] of files) {
+    await fs.copyFile(
+      join(rootDirectory, "remix.init", file),
+      join(rootDirectory, target || file)
+    );
+  }
+}
+
+async function updatePackageJsonForEdge(directory) {
+  const packageJson = await PackageJson.load(directory);
+  const {
+    dependencies: {
+      "@remix-run/netlify": _netlify,
+      "@remix-run/node": _node,
+      ...dependencies
+    },
+    scripts,
+    ...restOfPackageJson
+  } = packageJson.content;
+
+  packageJson.update({
+    // dev script is the same as the start script for Netlify Edge, "cross-env NODE_ENV=production netlify dev"
+    scripts: { ...scripts, dev: scripts["start"] },
+    ...restOfPackageJson,
+    dependencies: { ...dependencies, "@remix-run/netlify-edge": "*" },
+  });
+
+  await packageJson.save();
+}
+
+async function main({ rootDirectory }) {
```

Comment:
Removing remix.env.d.ts makes sense, but removing tsconfig.json causes errors in the editor. It'll still compile, but not a great experience potentially.

```
'React' refers to a UMD global, but the current file is a module. Consider adding an import instead.
```

Another thing to consider is if they create Netlify Functions (not Edge Functions), they may write them in TypeScript (probably since the project is TS already) and they may want the same settings.

From the Netlify blog post [Announcing Native TypeScript Support for Netlify Functions](https://www.netlify.com/blog/2021/04/19/announcing-native-typescript-support-for-netlify-functions/)

> We’ll automatically load any tsconfig.json configuration files found in your functions directory or the project’s root directory.

Thoughts @ascorbic?

Reply:
Thing that's still confusing me is my question I asked in https://github.com/remix-run/remix/pull/3104/files#r1034094380.
The answer to it, will probably resolve a lot of other questions/remarks